### PR TITLE
Simplify and standardise Makefile

### PR DESCRIPTION
**Pull request by nsoranzo**

### CODE REVIEW

**Comment by rcedgar** [comment]
Hi @nsoranzo thanks for this. @joelb123 suggested similar changes to the Makefile. The problem here is that the Makefiles (plural) in this project are generated automatically from Microsoft Visual Studio projects, so your proposed changes are not maintainable from my perspective. This means that the next time I post an update to the muscle repo, the Makefile will revert to its previous format -- better to be consistent IMO. I posted the script here: [https://github.com/rcedgar/vcxproj2makefile](https://github.com/rcedgar/vcxproj2makefile). If you can modify the python script to generate a "better" Makefile from a vcxproj file, that's an update I would consider. You could use the [muscle5.vcxproj](https://github.com/rcedgar/muscle/blob/main/src/muscle5.vcxproj) file as a test case.

**Comment by rcedgar** [comment]
Great, I see you understand Makefiles -- I find the syntax to be deeply obscure :-) Any chance you can also enable `ccache` if it's there? Fine if not, I'm up for adopting this and seeing how it goes.

**Comment by nsoranzo** [comment]
> Any chance you can also enable `ccache` if it's there? Fine if not, I'm up for adopting this and seeing how it goes.

It can be done, but I'd lean towards having the user enable `ccache` explicitly with `make CXX=ccache g++` (or they can define an alias for g++).

File: src/Makefile [hunk 1]
```diff
@@ -1,11 +1,12 @@
-CPP = g++
-CPPOPTS = -fopenmp -msse -mfpmath=sse -O3 -DNDEBUG -c
+UNAME_S := $(shell uname -s)
 
-CC = gcc
-CCOPTS = -fopenmp -msse -mfpmath=sse -O3 -DNDEBUG -c
+CXX = g++
```
Comment (mschubert) [inline src/Makefile:3]: Shouldn't this be set by `make` rather than the Makefile?

File: src/Makefile [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+CXXFLAGS = -O3 -DNDEBUG -fopenmp -ffast-math -msse -mfpmath=sse
```
Comment (mschubert) [inline src/Makefile:4]: Add `-std=c++11`? https://github.com/rcedgar/muscle/issues/4

**Comment by mschubert** [comment]
If I may add a comment here: As a package maintainer, it would be nice to have a Linux-compatible Makefile (i.e., this PR) also in addition to the one generated by VSCode (if you want to keep that)

**Pull request closed without merging**